packages/react-saasify/src/components/ServiceForm/ServiceForm.js

```diff
@@ -0,0 +1,77 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import theme from 'lib/theme'
+
+import { ReactFromJSONSchema } from './ReactFromJSONSchema'
+import { ServiceEntry } from './ServiceEntry'
+import { ServiceInput } from './ServiceInput'
+import { ServiceInputCheckbox } from './ServiceInputCheckbox'
+import { ServiceInputNumber } from './ServiceInputNumber'
+import { restrictDefinitionToExample, mergeInValues } from './helpers'
+
+import styles from './styles.module.css'
+
+export class ServiceForm extends Component {
+  static propTypes = {
+    onChange: PropTypes.func,
+    restrictToFirstExample: PropTypes.bool,
+    service: PropTypes.object
+  }
+
+  state = {
+    definition: {},
+    values: {}
+  }
+
+  static getDerivedStateFromProps(props, state) {
+    const definition = props.restrictToFirstExample
+      ? restrictDefinitionToExample(
+        props.service.definition,
+        props.service.examples[0],
+        state.values
+      )
+      : mergeInValues(props.service.definition)
+
+    return {
+      ...state,
+      definition
+    }
+  }
+
+  render() {
+    const { definition } = this.state
+
+    return (
+      <div className={theme(styles, 'service-form')}>
+        <ReactFromJSONSchema
+          definition={definition}
+          onChange={this._handleOnChange}
+          mapping={{
+            default: ({ _type }) => (
+              <div style={{ whiteSpace: 'nowrap' }}>
+                Missing input for {_type}
```

Comment:
I would hide this for now or figure out a way of disabling the playground if this is for a `required` property.

Reply:
Yes, this shouldn't have made its way in. Useful for debugging, though.

Reply:
Your call :)